### ISSUE TITLE
Function call gives bogus error about type arguments when real problem is a missing argument

### ISSUE THREAD

**Issue by mattmccutchen** [Microsoft/TypeScript#25683]
**TypeScript Version:**  master (60986ad)

**Search Terms:** "expected" "type arguments but got"

**Code**

```ts
function f<A, B = {}>(arg: number) {}
f<number>();
```

**Expected behavior:** Error: Expected 1 arguments, but got 0.

**Actual behavior:** Error: Expected 1-2 type arguments, but got 1.

**Playground Link:** [link](https://www.typescriptlang.org/play/#src=function%20f%3CA%2C%20B%20%3D%20%7B%7D%3E(arg%3A%20number)%20%7B%7D%0D%0Af%3Cnumber%3E()%3B)

**Related Issues:** #19374

**Comment by mattmccutchen** [comment]
Easy fix up at #25733.
